.github/workflows/main.yml

```diff
@@ -7,12 +7,26 @@ on:
   pull_request:
     types:
       - opened
+      - synchronize
 
 jobs:
   build:
     runs-on: self-hosted
     steps:
-      - name: Say Hello
-        run: echo "Hello, World!"
+      - name: nvidia-smi
+        run: nvidia-smi
 
+      - name: checkout repo
+        uses: actions/checkout@v4
 
+      - name: cmake configure
+        run: cmake -DCMAKE_BUILD_TYPE=Release -S gemm -B build .
+
+      - name: cmake build
+        run: cmake --build build
+
+      - name: google test
+        run: ctest --test-dir build --output-on-failure
+
+      - name: google benchmark
+        run: build/tests/bMatrix
```

Comment:
It's been a while since I've looked at GitHub workflows. Is this running all in one step?